### PR TITLE
add database pre process and update dev docker-compose config

### PR DESCRIPTION
**Pull request by NicolasMahe**
attached to epic https://github.com/liteflow-labs/company/issues/30

- add database pre process
- update dev docker-compose config to use new feature of https://github.com/liteflow-labs/backend/pull/143

how to test:
- ~~checkout backend@master~~
- ~~build the backend using the version v0.2: `docker build --tag liteflow/backend:v0.2 .`~~
- ~~update cli: comment line #100 of `dev.ts`:~~
image deployed on docker hub
- create processes like:
  - Get
```yml
name: store get
steps:
  - type: trigger
    endpointKey: get
  - type: task
    service: database
    taskKey: get
    inputs:
      collection: collectionOne
      key:
        key: body.key
```
  - list
```yml
name: store list
steps:
  - type: trigger
    endpointKey: list
  - type: task
    service: database
    taskKey: list
    inputs:
      collection: collectionOne
```
  - create
```yml
name: store create
steps:
  - type: trigger
    endpointKey: create
  - type: task
    service: database
    taskKey: create
    inputs:
      collection: collectionOne
      key:
        key: body.key
      value:
        key: body.value
```
  - update
```yml
name: store update
steps:
  - type: trigger
    endpointKey: update
  - type: task
    service: database
    taskKey: update
    inputs:
      collection: collectionOne
      key:
        key: body.key
      value:
        key: body.value
```
  - delete
```yml
name: store delete
steps:
  - type: trigger
    endpointKey: delete
  - type: task
    service: database
    taskKey: delete
    inputs:
      collection: collectionOne
      key:
        key: body.key
```
- run cli in dev command
- call endpoint like:
```
curl -XPOST -H "Content-Type: application/json" --data '{"collection":"user","value":{"foo":"bar"},"key":"one"}' http://localhost:3000/gateway/9f129da6-d41c-4847-a113-1bbd508f6bf6
```

### CODE REVIEW

File: packages/cli/src/utils/preprocess.ts [hunk 1]
```diff
@@ -31,3 +30,33 @@ export class FunctionPreProcess {
...
+            env: [
+              'POSTGRES_HOST=database',
+              'POSTGRES_PORT=5432',
+              'POSTGRES_USER=dev',
+              'POSTGRES_PASSWORD=dev',
+              'POSTGRES_DB=dev',
+            ],
+          },
+        }
+      : { instanceId: 'FIXME:' } // Current deployed version of service database
```
Comment (antho1404) [inline packages/cli/src/utils/preprocess.ts:54]: reminder to fix this

**Comment by antho1404** [comment]
To clarify, in the description you say to call the endpoint with `-H "Liteflow-Project: ea7a753b-0be4-4e6a-b6c1-235299f9af17"`. This should not be needed as the project is attached to the endpoint itself.

**Comment by NicolasMahe** [comment]
> To clarify, in the description you say to call the endpoint with `-H "Liteflow-Project: ea7a753b-0be4-4e6a-b6c1-235299f9af17"`. This should not be needed as the project is attached to the endpoint itself.

removed from curl command

File: packages/cli/src/utils/preprocess.ts (hunk 1 above)
Comment (NicolasMahe) [reply packages/cli/src/utils/preprocess.ts]: done

**Pull request merged**